src/python/pants/backend/jvm/tasks/coursier_resolve.py

```diff
@@ -587,6 +592,21 @@ def _map_coord_to_resolved_jars(cls, result, coursier_cache_path, pants_jar_path
 
     return coord_to_resolved_jars
 
+  @classmethod
+  def _get_path_to_jar(cls, coursier_cache_path, pants_jar_path_base, jar_path):
+    """
+    Create the path to the jar that will live in .pants.d
+    
+    :param coursier_cache_path: coursier cache location
+    :param pants_jar_path_base: location under pants workdir to store the symlink to the coursier cache
+    :param jar_path: path of the jar
+    :return:
+    """
+    if os.path.abspath(coursier_cache_path) not in os.path.abspath(jar_path):
+      return os.path.join(pants_jar_path_base, os.path.normpath('absolute/' + jar_path))
```

Comment:
Could you add the comment on the behavior of os.path.join which led to prepending `absolute/`?